plsc/entry.py

```diff
@@ -532,6 +532,39 @@ def _append_broadcast_ops(self, program):
                 outputs={'Out': var},
                 attrs={'use_calc_stream': True})
 
+    def _sync_program(self):
+        """ Use allreduce operation to sync all trainers. """
```

Comment:
too complicated. Could a single program with allreduce or broadcast be enough?

Reply:
done.